// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
import "google/protobuf/empty.proto";
package skrcompiler;

// The greeting service definition.
service CompileResource
{
    // Sends a greeting
    rpc Compile(CompileInfo) returns (CompileResult) {}
}

service HostResource
{
    rpc GetPath(ResourceId) returns (ResourcePath) {}
    rpc Register(Port) returns (google.protobuf.Empty) {}
}

// The request message containing the user's name.
message CompileInfo
{
    string json = 1;
}

// The response message containing the greetings
message CompileResult
{
    int32 errorCode = 1;
    string errorMessage = 2;
}

message Port
{
    string number = 1;
}

message ResourceId
{
    string guid = 1;
}

message ResourcePath
{
    string path = 1;
}